src/core/instances.jl

```diff
@@ -168,10 +168,33 @@ function Base.getindex(obj::AbstractCompositeComponentInstance, comp_name::Symbo
     return compinstance(obj, comp_name)
 end
 
-function _get_datum(ci::AbstractComponentInstance, datum_name::Symbol)
+# LFR TODO some of this is very similar to _get_datum for leaf component instance, 
+# but in some ways it seems clearner to separate them
+function _get_datum(ci::CompositeComponentInstance, datum_name::Symbol)
     vars = variables(ci)
 
-    if datum_name in names(vars)
+    if datum_name in keys(vars) # could merge with below if made names(NamedTuple) = keys(NamedTuple)
+        which = vars
+    else
+        pars = parameters(ci)
+        if datum_name in keys(pars) # could merge with below if made names(NamedTuple) = keys(NamedTuple)
+            which = pars
+        else
+            error("$datum_name is not a parameter or a variable in component $(ci.comp_path).")
+        end
+    end
+    
+    ref = getproperty(which, datum_name)
+    
+    # LFR TODO get the comp_name and datum_name to use ...
```

Comment:
@ckingdon95 this is a little bit tricky, because sometimes `UnnamedReference` skips a layer, so the `ref.comp_name` is not held in the `ci.comps_dict` ie. in the test we have currently we have a `CompositeComponentInstance` where parameters are fine:
```
julia> ci.parameters 
(fooA1 = Mimi.UnnamedReference(:A, :foo1), 
fooA2 = Mimi.UnnamedReference(:A, :foo2), 
foo3 = Mimi.UnnamedReference(:B, :foo3),
foo4 = Mimi.UnnamedReference(:B, :foo4), 
par_1_1 = Mimi.UnnamedReference(:A, :par_1_1))
```
but the variable just goes all the way to `:Comp3`
```
julia> ci.variables
(var_3_1 = Mimi.UnnamedReference(:Comp3, :var_3_1),)
```
and thus calling `_get_datum(ref.comp_name, ref.datum_name)` won't work...